web-common/src/features/dashboards/time-controls/TimeRangeSelector.svelte

```diff
@@ -99,195 +108,196 @@
-  function onCalendarClose() {
-    isCalendarRecentlyClosed = true;
-
-    setTimeout(() => {
-      isCalendarRecentlyClosed = false;
-    }, 300);
+  function toggleFloatingElement() {
+    open = !open;
   }
-
-  $: currentSelection = $dashboardStore?.selectedTimeRange?.name;
-  $: intermediateSelection = currentSelection;
-
-  const handleMenuOpen = () => {
-    if (intermediateSelection !== TimeRangePreset.CUSTOM) {
-      isCustomRangeOpen = false;
-    }
-  };
 </script>
 
-<WithTogglableFloatingElement
-  alignment="start"
-  distance={8}
-  let:active
-  let:toggleFloatingElement
-  on:open={handleMenuOpen}
+<DropdownMenu.Root
+  bind:open
+  onOpenChange={handleMenuOpen}
+  closeOnItemClick={false}
 >
-  {#if hasSubRangeSelected}
-    <div class="flex">
-      <TimeRangeScrubChip
-        on:click={toggleFloatingElement}
-        on:remove={() => dispatch("remove-scrub")}
-        {active}
-        start={$dashboardStore?.selectedScrubRange?.start}
-        end={$dashboardStore?.selectedScrubRange?.end}
-        zone={$dashboardStore?.selectedTimezone}
-      />
-    </div>
-  {:else}
-    <button
-      class:bg-gray-200={active}
-      class="px-3 py-2 rounded flex flex-row gap-x-2 hover:bg-gray-200 hover:dark:bg-gray-600 items-baseline"
-      on:click={toggleFloatingElement}
-      aria-label="Select time range"
-    >
-      <div class="flex flew-row gap-x-3">
-        <div class="font-bold flex flex-row items-center gap-x-3">
-          <span class="ui-copy-icon"><Calendar size="16px" /></span>
-          <span style:transform="translateY(1px)">
-            <!-- This conditional shouldn't be necessary because there should always be a selected (at least default) time range -->
-            {#if intermediateSelection === TimeRangePreset.CUSTOM}
-              Custom range
-            {:else if currentSelection}
-              {#if currentSelection in DEFAULT_TIME_RANGES}
-                {DEFAULT_TIME_RANGES[currentSelection].label}
-              {:else}
-                Last {humaniseISODuration(currentSelection)}
-              {/if}
+  <DropdownMenu.Trigger asChild let:builder>
+    {#if scrubStart && scrubEnd && timeZone}
+      <div class="flex" use:builder.action {...builder}>
+        <TimeRangeScrubChip
+          on:remove={() => dispatch("remove-scrub")}
+          active={open}
+          start={scrubStart}
+          end={scrubEnd}
+          zone={timeZone}
+        />
+      </div>
+    {:else}
+      <button
+        use:builder.action
+        {...builder}
+        class:bg-gray-200={open}
+        class="flex items-center gap-x-2 rounded px-3 py-2 hover:bg-gray-200 hover:dark:bg-gray-600"
+        aria-label="Select time range"
+      >
+        <span class="ui-copy-icon"><Calendar size="16px" /></span>
+        <b>
+          <!-- This conditional shouldn't be necessary because there should always be a selected (at least default) time range -->
```

Comment:
What does this comment mean? It certainly looks like this conditional is needed... is this comment perhaps no longer accurate?

Not a blocker cuz I know you didn't leave this here, but I guess as long as you're digging around in this code, it would be nice to investigate what this might have meant, and if it's no longer correct, remove the comment, or if it is correct, clean out the conditional...